ghcide/src/Development/IDE/Spans/Documentation.hs

```diff
@@ -88,19 +88,21 @@ getDocumentationsTryGhc env mod names = do
         (docFu, srcFu) <-
             case nameModule_maybe name of
             Just mod -> liftIO $ do
-                doc <- toFileUriText $ lookupDocHtmlForModule env mod
-                src <- toFileUriText $ lookupSrcHtmlForModule env mod
+                let
+                  toUriFileText :: (HscEnv -> Module -> IO (Maybe FilePath)) -> IO (Maybe T.Text)
+                  toUriFileText f = (fmap . fmap) (getUri . filePathToUri) $ f env mod
+                doc <- toUriFileText lookupDocHtmlForModule
+                src <- toUriFileText lookupSrcHtmlForModule
                 return (doc, src)
             Nothing -> pure mempty
-        let docUri = (<> "#" <> selector <> showNameWithoutUniques name) <$> docFu
-            srcUri = (<> "#" <> showNameWithoutUniques name) <$> srcFu
-            selector
-                | isValName name = "v:"
-                | otherwise = "t:"
+        let
+            embelishUri :: Functor f => T.Text -> f T.Text -> f T.Text
+            embelishUri f = fmap (<> "#" <> f <> showNameWithoutUniques name)
+
+            docUri = embelishUri (bool "t:" "v:" $ isValName name) docFu
+            srcUri = embelishUri mempty srcFu
+
         return $ SpanDocUris docUri srcUri
-       where
-        toFileUriText :: IO (Maybe FilePath) -> IO (Maybe T.Text)
-        toFileUriText = (fmap . fmap) (getUri . filePathToUri)
```

Comment:
Pulling this into the local scope of `do` it is used in & bundling shared args (`env` & `mod`) into it.